pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -398,3 +416,97 @@ func (e *reconcileError) reason() string {
 func newReconcileError(reason string, err error) *reconcileError {
 	return &reconcileError{degradationReason: reason, err: err}
 }
+
+// generatePublicKey generates a public key and returns it in the ssh-rsa..
+// string format using the retrieved private key
+func generatePublicKey() (string, error) {
+	privateKey, err := retrievePrivateKey()
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve private key in rsa format: %v", err)
+	}
+	publicKey, err := ssh.NewPublicKey(&privateKey.PublicKey)
+	if err != nil {
+		return "", fmt.Errorf("failed to generate public key: %v", err)
+	}
+	pubKeyBytes := ssh.MarshalAuthorizedKey(publicKey)
+	return string(pubKeyBytes[:]), nil
+}
+
+// retrievePrivateKey retrieves private key in format required for generating
+// public key using the privateKeyPath
```

Comment:
/retrievePrivateKey retries the private key from the privateKeyPath in a format required for generating the public key`